zebra-state/src/service/finalized_state.rs

```diff
@@ -181,19 +131,34 @@ impl FinalizedState {
         self.tip().map(|(height, _)| height)
     }
 
+    fn is_empty(&self, cf: &rocksdb::ColumnFamily) -> bool {
+        // use iterator to check if it's empty
+        !self
+            .db
+            .iterator_cf(cf, rocksdb::IteratorMode::Start)
+            .valid()
+    }
+
     /// Immediately commit `block` to the finalized state.
     pub fn commit_finalized_direct(&mut self, block: Arc<Block>) -> Result<block::Hash, BoxError> {
-        use sled::Transactional;
-
         let height = block
             .coinbase_height()
             .expect("finalized blocks are valid and have a coinbase height");
         let hash = block.hash();
 
         block_precommit_metrics(&hash, height, &block);
+        tracing::trace!(?height, ?hash, "Finalized block");
```

Comment:
```suggestion
```
I think this is redundant because `block_precommit_metrics` puts a more detailed event